.github/workflows/build-wheels.yml

```
@@ -53,5 +67,26 @@ jobs:
       - name: Upload wheels to GH
         uses: actions/upload-artifact@v4
         with:
-          name: cibw-wheels-${{ matrix.os }}
+          name: cibw-wheels-${{ matrix.runs-on }}
           path: wheelhouse/*.whl
+
+  pypi-publish:
+    name: Upload release to PyPI
+    needs: [make_sdist, build_wheels]
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'published'
+    environment:
+      name: pypi
+      url: https://pypi.org/p/Wasserstein
+    permissions:
+      id-token: write
+    steps:
+      - name: Download wheels from GH
+        uses: actions/download-artifact@v4
+        with:
+          pattern: cibw-*
+          path: dist
+          merge-multiple: true
+
+      - name: Upload package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
```

```suggestion
        uses: pypa/gh-action-pypi-publish@release/v1
        with:
          print-hash: true
          attestations: true
```

This just adds some additional options from https://github.com/pypa/gh-action-pypi-publish.

---

You could also add artifact generation. You'd also need `attestations: write` permission for that. But that could all be in a followup?